test/rubygems/test_gem_installer.rb

```diff
@@ -1222,6 +1230,35 @@ def test_install_creates_binstub_that_dont_trust_encoding
     end
 
     assert_match(/ran executable/, e.message)
+
+    assert_path_not_exist(File.join(installer.bin_dir, "executable.lock"))
+  end
+
+  def test_install_does_not_leave_lockfile_for_binstub
+    installer = util_setup_installer
+
+    installer.wrappers = true
+
+    File.class_eval do
+      alias_method :original_chmod, :chmod
+      define_method(:chmod) do |mode|
+        original_chmod(mode)
+        raise Gem::Ext::BuildError if path.end_with?("/executable")
+      end
+    end
+
+    assert_raise(Gem::Ext::BuildError) do
+      installer.install
+    end
+
+    assert_path_not_exist(File.join(installer.bin_dir, "executable.lock"))
+    # assert_path_not_exist(File.join(installer.bin_dir, "executable"))
```

Comment:
Did you mean to uncomment this? or is it not passing and you left it there as a TODO? If the latter, can you add a comment on top of it explaining it?

Reply:
I had to confirm it because I was uncertain if executables would be removed or left when an installer failed in middle.
Currently it seems to leave already copied files in that case.
Is this behavior expected or a TODO?

Reply:
In my opinion that's a bug that we should fix. A aborted installer would ideally leave no side-effects?

Reply:
Updated.

Reply:
The `executable` seems to be removed on JRuby and TruffleRuby...